src/pages/task/actionButtons/PreviousCommits.tsx

```diff
@@ -0,0 +1,212 @@
+    if (versionMetadata?.isPatch === false) {
+      fetchParentTask({
+        variables: {
+          projectIdentifier,
+          skipOrderNumber,
+          buildVariantOptions: {
+            ...bvOptionsBase,
+          },
+        },
+      });
+    } else if (versionMetadata?.isPatch === true) {
+      setParentTask(baseTask);
+    }
+  }, [versionMetadata]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Hook to find last passing & last executed tasks. This is a separate hook because the parent task must be
+  // known before proceeding.
+  useEffect(() => {
+    if (parentTask) {
+      if (parentTask.status !== TaskStatus.Succeeded) {
+        fetchLastPassing({
+          variables: {
+            projectIdentifier,
+            skipOrderNumber,
+            buildVariantOptions: {
+              ...bvOptionsBase,
+              statuses: [TaskStatus.Succeeded],
+            },
+          },
+        });
+      }
+      if (!isFinishedTaskStatus(parentTask.status)) {
+        fetchLastExecuted({
+          variables: {
+            projectIdentifier,
+            skipOrderNumber,
+            buildVariantOptions: {
+              ...bvOptionsBase,
+              statuses: finishedTaskStatuses,
+            },
+          },
+        });
+      }
+    }
+  }, [parentTask]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // The default link should be undefined so that the GO button is disabled if links do not exist.
+  let link: string;
+  if (parentTask) {
+    switch (selectState) {
+      case CommitType.Base:
+        link = getTaskRoute(parentTask.id);
+        break;
+      // If a base task succeeded, the last passing commit is the base task.
+      case CommitType.LastPassing:
+        link = getTaskRoute(lastPassingTask?.id || parentTask.id);
+        break;
+      // If a base task exists and has finished, the last executed commit is the base task.
+      case CommitType.LastExecuted:
+        link = getTaskRoute(lastExecutedTask?.id || parentTask.id);
+        break;
+      default:
+    }
+  }
+
+  return versionMetadata?.isPatch !== undefined ? (
+    <PreviousCommitsWrapper>
+      <StyledSelect
+        size="small"
+        data-cy="previous-commits"
+        label="Previous commits for this task"
+        allowDeselect={false}
+        onChange={(v: CommitType) => setSelectState(v)}
+        value={selectState}
+        disabled={!versionMetadata?.baseVersion}
+      >
+        <Option value={CommitType.Base} key={CommitType.Base}>
+          Go to {versionMetadata?.isPatch ? "base commit" : "parent commit"}
+        </Option>
+        <Option value={CommitType.LastPassing} key={CommitType.LastPassing}>
+          Go to last passing version
+        </Option>
+        <Option value={CommitType.LastExecuted} key={CommitType.LastExecuted}>
+          Go to last executed version
+        </Option>
+      </StyledSelect>
+      <Button as={Link} to={link || "/"} disabled={!link} size="small">
```

Comment:
It's to prevent this error in the console:

<img width="457" alt="Screen Shot 2022-02-15 at 11 06 48 AM" src="https://user-images.githubusercontent.com/47064971/154101217-40c892a4-cbd7-45e4-9403-79be0e073a60.png">

Reply:
Resolving because this code changed and went away